miden-tx/src/tests/kernel_tests/test_prologue.rs

```diff
@@ -99,127 +99,127 @@ fn test_transaction_prologue() {
     );
 
     tx_context.set_tx_args(tx_args);
-    let process = tx_context.execute_code(code).unwrap();
+    let process = &tx_context.execute_code(code).unwrap();
 
-    global_input_memory_assertions(&process, &tx_context);
-    block_data_memory_assertions(&process, &tx_context);
-    chain_mmr_memory_assertions(&process, &tx_context);
-    account_data_memory_assertions(&process, &tx_context);
-    input_notes_memory_assertions(&process, &tx_context, &note_args);
+    global_input_memory_assertions(process, &tx_context);
+    block_data_memory_assertions(process, &tx_context);
+    chain_mmr_memory_assertions(process, &tx_context);
+    account_data_memory_assertions(process, &tx_context);
+    input_notes_memory_assertions(process, &tx_context, &note_args);
 }
 
 fn global_input_memory_assertions(process: &Process, inputs: &TransactionContext) {
     assert_eq!(
-        read_root_mem_value(&process.into(), BLK_HASH_PTR),
+        read_root_mem_word(&process.into(), BLK_HASH_PTR),
```

Comment:
For `global_input_memory_assertions` or `read_root_mem_word` and similar functions we could consider taking `process_state: impl Into<ProcessState<'_>>`. Not 100% sure if this works without an explicit lifetime, but if we can then pass a `&Process`, that would be a nice API.
Since you already updated all call sites now, that's optional - we can also do this in the future or not at all.